buildSrc/src/main/java/org/opensearch/gradle/RepositoriesSetupPlugin.java

```diff
@@ -92,10 +92,9 @@ public static void configureRepositories(Project project) {
                 throw new GradleException("Malformed lucene snapshot version: " + luceneVersion);
             }
             String revision = matcher.group(1);
-            // TODO(cleanup) - Setup own lucene snapshot repo
             MavenArtifactRepository luceneRepo = repos.maven(repo -> {
                 repo.setName("lucene-snapshots");
-                repo.setUrl("https://artifacts.opensearch.org/snapshots/lucene/");
+                repo.setUrl("https://d1nvenhzbhpy0q.cloudfront.net/snapshots/lucene/");
```

Comment:
I believe it should be `https://aws.oss.sonatype.org/content/repositories/snapshot`, that is the convention OpenSearch relies on ...

Reply:
Hey @reta thanks for the review.  The workflow here is building and pushing these snapshots to a "maven" repo that is simply an S3 bucket served by a cloudfront distribution we see here, not the sonatype repo.

I've done this for a few reasons.  First, is that these are not snapshots of OpenSearch artifacts themselves, but a component we use to test with.  They are not intended to be consumed by other repositories and be confused with [official lucene snapshots](https://lucene.apache.org/core/developer.html).   We can't use these for the limitations outlined [here](https://github.com/opensearch-project/opensearch-build/issues/1778#issuecomment-1072661738).  

Second, We won't have the ability to push these to sonatype because they are not under the org.opensearch namespace, so we'd have to rewrite the group Id in order to do so, which personally I thought was overkill for what we are trying to use them for and adds unneeded complexity when publishing and consuming in RepositoriesSetupPlugin.

Reply:
And what's nice is that any maintainer can manually invoke the workflow to push latest lucene snapshots so we can quickly upgrade the working branches w/o long tail changes.

Reply:
:+1: thanks @mch2 !

Reply:
One reason to change `d1nvenhzbhpy0q.cloudfront.net` for `ci.opensearch.org/lucene/snapshots` would be readability. Everything else makes complete sense.